bin/rebuild-package-locks.js

```diff
@@ -13,47 +13,69 @@
 const path = require('path');
 const fs = require('fs-extra');
 const Project = require('@lerna/project');
+const filterPackages = require('@lerna/filter-packages');
 const build = require('../packages/build');
 
 /**
  * Remove all package-lock.json and node_modules for all packages
  * @param {Project} project The lerna project
+ * @param {string[]} scopes A list of package names to be included
  */
-async function removePackageLocks(project) {
+async function removePackageLocks(project, ...scopes) {
   const packages = await project.getPackages();
   const rootPath = project.rootPath;
   const pkgRoots = [];
-  for (const pkg of packages) {
+  const matchedPackages = filterPackages(packages, scopes, [], true, true);
+  if (matchedPackages.length === 0) return;
+  for (const pkg of matchedPackages) {
     pkgRoots.push(pkg.location);
   }
-  pkgRoots.push(rootPath);
+  // Only clean the monorepo root package if no scopes is provided
+  if (scopes.length === 0) pkgRoots.push(rootPath);
 
   console.log('Cleaning package-lock.json and node_modules...');
   await Promise.all(
     pkgRoots.map(async root => {
+      console.log('  - %s', path.relative(rootPath, root));
       await fs.remove(path.join(root, 'package-lock.json'));
       await fs.remove(path.join(root, 'node_modules'));
     }),
   );
+  return pkgRoots;
 }
 
 /**
  * Rebuild package-lock.json files:
  *
- * 1. Remove node_modules and package-lock.json for all packages, including the
- * root one
- * 2. Run `npm install` to regenerate package-lock.json files
+ * 1. Remove node_modules and package-lock.json for matching packages,
+ * including the root one if no scopes is specified
+ * 2. Run `npx lerna bootstrap or npm install` to regenerate package-lock.json
+ * files
```

Comment:
I am not a big fan of updating package lock files during our release process.
- I don't want the release commit to be cluttered with changes in package-lock files. 
- Also an update of a (deep) dependency may introduce a bug. This bug could be platform-specific, local `npm test` may not catch it. IMO, we should always test package-lock updates via our CI. 
- If the bug does not manifest in our CI env but only later in user environments, then the task of reverting the package-lock update becomes tricky, as the package-lock changes are coupled with the new version release. 

IMO, update of package-lock files should be a standalone commit.

We are using RenovateBot to update the lock files. As long as we merge all outstanding RenovateBot PRs before starting the release process, then I believe there is no need to touch package locks in our release process.

If we think it's important to touch package locks as part of the release process, then please make sure those changes are committed separately from version-specific changes.

Reply:
I agree. The `build:full` script is now changed to reflect CI behavior:

```
"build:full": "npm ci --ignore-scripts && lerna bootstrap --ci && npm run clean && npm run build",
```

It basically run `npm ci` for all packages to establish a clean and frozen state for all dependencies. Then a clean TSC build is performed.

Reply:
Sounds good. Can you please update this comment to match the actual implementation?

Reply:
The comment matches the implementation of `rebuild-package-locks`. The release script does not use it to have a clean build.